unsigned int tomo::IoLoop::epoll_pass<false, false, false>(int)
--- PRE

+++ POST

@@ -2,7 +2,7 @@

 0004 [ 2] push %r15
 0006 [ 2] push %r14
 0008 [ 2] push %r13
-000a [ 3] mov %rdi,%r13
+000a [ 3] mov %rdi,%r15
 000d [ 2] push %r12
 000f [ 1] push %rbp
 0010 [ 1] push %rbx
@@ -13,171 +13,157 @@

 0024 [ 5] mov $0x100,%edx
 0029 [ 3] mov %rbx,%rsi
 002c [ 3] xor %r12d,%r12d
-002f [ 6] mov $0x0,%r14d
+002f [ 6] mov $0x0,%r13d
 0035 [ 9] mov %fs:0x28,%rax
 003e [ 5] mov %rax,0x28(%rsp)
 0043 [ 2] xor %eax,%eax
 0045 [ 5] call <epoll_wait@plt>
 004a [ 2] test %eax,%eax
 004c [ 4] cmovns %eax,%r12d
-0050 [ 7] mov %r12d,0xc18(%r13)
-0057 [ 6] jle <+0x11f>
-005d [ 7] mov 0x190(%r13),%rax
-0064 [ 3] movslq %r12d,%rdx
-0067 [ 2] xor %ebp,%ebp
-0069 [ 7] lea <rip>(%rip),%r15 # <local-switch-0x110,0x250,0x1d0,0x110,0x110,0x180,0x110,0x110,0x100,0x110,0x110,0x110,0x110,0x110,0x110,0x110,0x110,0x150>
-0070 [ 7] add %rdx,0x520(%rax)
-0077 [ 7] lea 0xac8(%r13),%rax
-007e [ 5] mov %rax,0x8(%rsp)
-0083 [ 5] lea 0x20(%rsp),%rax
-0088 [ 4] mov %rax,(%rsp)
-008c [ 4] nopl 0x0(%rax)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
+0050 [ 7] mov %r12d,0xc18(%r15)
+0057 [ 2] jle <+0xcb>
+0059 [ 7] mov 0x190(%r15),%rax
+0060 [ 3] movslq %r12d,%rdx
+0063 [ 2] xor %ebp,%ebp
+0065 [ 7] lea <rip>(%rip),%r14 # <local-switch-0xb0,0x240,0x100,0xb0,0xb0,0x1f0,0xc0,0xb0,0x1b0,0xc0,0xb0,0xb0,0xb0,0xb0,0xb0,0xb0,0xb0,0x1c0>
+006c [ 7] add %rdx,0x520(%rax)
+0073 [ 7] lea 0xac8(%r15),%rax
+007a [ 5] mov %rax,0x8(%rsp)
+007f [ 5] lea 0x20(%rsp),%rax
+0084 [ 4] mov %rax,(%rsp)
+0088 [ 8] nopl 0x0(%rax,%rax,1)
+0090 [ 4] mov 0x4(%rbx),%rsi
+0094 [ 3] mov %rsi,%rax
 0097 [ 4] shr $0x38,%rax
 009b [ 2] cmp $0x11,%al
-009d [ 2] ja <+0x110>
-009f [ 4] movslq (%r15,%rax,4),%rax
-00a3 [ 3] add %r15,%rax
+009d [ 2] ja <+0xb0>
+009f [ 4] movslq (%r14,%rax,4),%rax
+00a3 [ 3] add %r14,%rax
 00a6 [ 3] notrack jmp *%rax
-00a9 [ 2] mov (%rbx),%eax
-00ab [ 5] test $0x2019,%eax
-00b0 [ 2] je <+0xb6>
-00b2 [ 4] movb $0x0,0x29(%rdx)
-00b6 [ 2] test $0x1c,%al
-00b8 [ 2] je <+0xf2>
-00ba [ 4] cmpb $0x0,0x30(%rdx)
-00be [ 5] mov %rdx,0x20(%rsp)
-00c3 [ 2] jne <+0xf2>
-00c5 [ 7] mov 0x1d8(%r13),%rcx
-00cc [ 7] mov 0x1c8(%r13),%rax
-00d3 [ 4] movb $0x1,0x30(%rdx)
-00d7 [ 4] sub $0x8,%rcx
-00db [ 3] cmp %rcx,%rax
-00de [ 6] je <+0x2fa>
-00e4 [ 3] mov %rdx,(%rax)
-00e7 [ 4] add $0x8,%rax
-00eb [ 7] mov %rax,0x1c8(%r13)
-00f2 [ 3] mov %rdx,%rsi
-00f5 [ 3] mov %r13,%rdi
-00f8 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
-00fd [ 3] nopl (%rax)
-0100 [ 3] inc %r14d
-0103 [11] data16 cs nopw 0x0(%rax,%rax,1)
-010e [ 2] xchg %ax,%ax
-0110 [ 2] inc %ebp
-0112 [ 4] add $0xc,%rbx
-0116 [ 3] cmp %r12d,%ebp
-0119 [ 6] jl <+0x90>
-011f [ 5] mov 0x28(%rsp),%rax
-0124 [ 9] sub %fs:0x28,%rax
-012d [ 6] jne <+0x2f5>
-0133 [ 4] add $0x38,%rsp
-0137 [ 3] mov %r14d,%eax
-013a [ 1] pop %rbx
-013b [ 1] pop %rbp
-013c [ 2] pop %r12
-013e [ 2] pop %r13
-0140 [ 2] pop %r14
-0142 [ 2] pop %r15
-0144 [ 1] ret
-0145 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0150 [ 7] mov 0x190(%r13),%rax
-0157 [ 7] mov 0x188(%r13),%rdi
-015e [ 5] mov 0x8(%rsp),%rdx
-0163 [ 2] inc %ebp
-0165 [ 3] inc %r14d
-0168 [ 4] add $0xc,%rbx
-016c [ 2] mov (%rax),%esi
-016e [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
-0173 [ 3] cmp %r12d,%ebp
-0176 [ 6] jl <+0x90>
-017c [ 2] jmp <+0x11f>
-017e [ 2] xchg %ax,%ax
-0180 [ 9] movq $0x0,0x20(%rsp)
-0189 [ 7] nopl 0x0(%rax)
-0190 [ 7] mov 0xbc8(%r13),%edi
-0197 [ 4] mov (%rsp),%rsi
-019b [ 5] mov $0x8,%edx
-01a0 [ 5] call <read@plt>
-01a5 [ 4] cmp $0x8,%rax
-01a9 [ 2] je <+0x190>
-01ab [ 7] mov 0x190(%r13),%rax
-01b2 [ 2] inc %ebp
-01b4 [ 3] inc %r14d
-01b7 [ 4] add $0xc,%rbx
-01bb [ 7] incq 0x440(%rax)
-01c2 [ 3] cmp %r12d,%ebp
-01c5 [ 6] jl <+0x90>
-01cb [ 5] jmp <+0x11f>
-01d0 [10] movabs $0xffffffffffff,%rax
-01da [ 3] and %rax,%rdx
-01dd [ 6] je <+0x110>
-01e3 [ 6] mov 0x788(%rdx),%eax
-01e9 [ 7] mov 0x190(%r13),%rcx
-01f0 [ 2] cmp %eax,(%rcx)
-01f2 [ 6] jne <+0x110>
-01f8 [ 4] cmpb $0x0,0x33(%rdx)
-01fc [ 6] jne <+0x110>
-0202 [ 6] mov 0x784(%rdx),%eax
-0208 [ 3] cmp $0xfffffffe,%eax
-020b [ 6] je <+0x110>
-0211 [ 7] mov 0x288(%r13),%rax
-0218 [ 7] mov 0x290(%r13),%rcx
-021f [ 3] cmp %rcx,%rax
-0222 [ 2] jne <+0x23d>
-0224 [ 5] jmp <+0xa9>
-0229 [ 7] nopl 0x0(%rax)
-0230 [ 4] add $0x28,%rax
-0234 [ 3] cmp %rax,%rcx
-0237 [ 6] je <+0xa9>
-023d [ 3] cmp (%rax),%rdx
-0240 [ 2] jne <+0x230>
-0242 [ 5] jmp <+0x110>
-0247 [ 9] nopw 0x0(%rax,%rax,1)
-0250 [ 7] mov 0x188(%r13),%rax
-0257 [ 7] movzbl 0x115e0(%rax),%eax
-025e [ 2] cmp $0x3,%al
-0260 [ 6] ja <+0x110>
-0266 [ 7] mov 0xaa0(%r13),%eax
-026d [ 4] mov %eax,0x10(%rsp)
-0271 [ 2] test %eax,%eax
-0273 [ 6] js <+0x110>
-0279 [ 8] movl $0x0,0x1c(%rsp)
-0281 [ 4] mov 0x10(%rsp),%edi
-0285 [ 2] xor %edx,%edx
-0287 [ 2] xor %esi,%esi
-0289 [ 5] mov $0x80800,%ecx
-028e [ 5] call <accept4@plt>
-0293 [ 2] test %eax,%eax
-0295 [ 2] jns <+0x2e0>
-0297 [ 5] call <__errno_location@plt>
-029c [ 2] mov (%rax),%eax
-029e [ 3] cmp $0x4,%eax
-02a1 [ 2] je <+0x281>
-02a3 [ 3] cmp $0xb,%eax
-02a6 [ 2] je <+0x2b6>
-02a8 [ 7] mov 0x190(%r13),%rax
-02af [ 7] incq 0x458(%rax)
-02b6 [ 4] mov 0x1c(%rsp),%eax
-02ba [ 2] inc %ebp
-02bc [ 4] add $0xc,%rbx
-02c0 [ 3] add %eax,%r14d
-02c3 [ 3] cmp %r12d,%ebp
-02c6 [ 6] jl <+0x90>
-02cc [ 5] jmp <+0x11f>
-02d1 [11] data16 cs nopw 0x0(%rax,%rax,1)
-02dc [ 4] nopl 0x0(%rax)
-02e0 [ 5] mov $0x1,%edx
-02e5 [ 2] mov %eax,%esi
-02e7 [ 3] mov %r13,%rdi
-02ea [ 4] incl 0x1c(%rsp)
-02ee [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
-02f3 [ 2] jmp <+0x281>
-02f5 [ 5] call <__stack_chk_fail@plt>
-02fa [ 5] lea 0x20(%rsp),%rsi
-02ff [ 7] lea 0x198(%r13),%rdi
-0306 [ 5] mov %rdx,0x10(%rsp)
-030b [ 5] call <void std::deque<tomo::Client*, std::allocator<tomo::Client*> >::_M_push_back_aux<tomo::Client* const&>(tomo::Client* const&)>
-0310 [ 5] mov 0x10(%rsp),%rdx
-0315 [ 5] jmp <+0xf2>
+00a9 [ 7] nopl 0x0(%rax)
+00b0 [ 3] mov %r15,%rdi
+00b3 [ 5] call <tomo::IoLoop::fliplisten_epoll(unsigned long)>
+00b8 [ 3] add %eax,%r13d
+00bb [ 5] nopl 0x0(%rax,%rax,1)
+00c0 [ 2] inc %ebp
+00c2 [ 4] add $0xc,%rbx
+00c6 [ 3] cmp %r12d,%ebp
+00c9 [ 2] jl <+0x90>
+00cb [ 5] mov 0x28(%rsp),%rax
+00d0 [ 9] sub %fs:0x28,%rax
+00d9 [ 6] jne <+0x2d5>
+00df [ 4] add $0x38,%rsp
+00e3 [ 3] mov %r13d,%eax
+00e6 [ 1] pop %rbx
+00e7 [ 1] pop %rbp
+00e8 [ 2] pop %r12
+00ea [ 2] pop %r13
+00ec [ 2] pop %r14
+00ee [ 2] pop %r15
+00f0 [ 1] ret
+00f1 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00fc [ 4] nopl 0x0(%rax)
+0100 [10] movabs $0xffffffffffff,%rax
+010a [ 3] and %rsi,%rax
+010d [ 2] je <+0xc0>
+010f [ 6] mov 0x788(%rax),%edx
+0115 [ 7] mov 0x190(%r15),%rcx
+011c [ 2] cmp %edx,(%rcx)
+011e [ 2] jne <+0xc0>
+0120 [ 4] cmpb $0x0,0x33(%rax)
+0124 [ 2] jne <+0xc0>
+0126 [ 6] mov 0x784(%rax),%edx
+012c [ 3] cmp $0xfffffffe,%edx
+012f [ 2] je <+0xc0>
+0131 [ 7] mov 0x288(%r15),%rdx
+0138 [ 7] mov 0x290(%r15),%rcx
+013f [ 3] cmp %rcx,%rdx
+0142 [ 2] je <+0x162>
+0144 [11] data16 cs nopw 0x0(%rax,%rax,1)
+014f [ 1] nop
+0150 [ 3] cmp (%rdx),%rax
+0153 [ 6] je <+0xc0>
+0159 [ 4] add $0x28,%rdx
+015d [ 3] cmp %rdx,%rcx
+0160 [ 2] jne <+0x150>
+0162 [ 2] mov (%rbx),%edx
+0164 [ 6] test $0x2019,%edx
+016a [ 2] je <+0x170>
+016c [ 4] movb $0x0,0x29(%rax)
+0170 [ 3] and $0x1c,%edx
+0173 [ 2] je <+0x19f>
+0175 [ 4] cmpb $0x0,0x30(%rax)
+0179 [ 5] mov %rax,0x20(%rsp)
+017e [ 2] jne <+0x19f>
+0180 [ 5] lea 0x20(%rsp),%rsi
+0185 [ 7] lea 0x198(%r15),%rdi
+018c [ 4] movb $0x1,0x30(%rax)
+0190 [ 5] mov %rax,0x10(%rsp)
+0195 [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
+019a [ 5] mov 0x10(%rsp),%rax
+019f [ 3] mov %rax,%rsi
+01a2 [ 3] mov %r15,%rdi
+01a5 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
+01aa [ 6] nopw 0x0(%rax,%rax,1)
+01b0 [ 3] inc %r13d
+01b3 [ 5] jmp <+0xc0>
+01b8 [ 8] nopl 0x0(%rax,%rax,1)
+01c0 [ 7] mov 0x190(%r15),%rax
+01c7 [ 7] mov 0x188(%r15),%rdi
+01ce [ 5] mov 0x8(%rsp),%rdx
+01d3 [ 3] inc %r13d
+01d6 [ 2] mov (%rax),%esi
+01d8 [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
+01dd [ 5] jmp <+0xc0>
+01e2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+01ed [ 3] nopl (%rax)
+01f0 [ 9] movq $0x0,0x20(%rsp)
+01f9 [ 7] nopl 0x0(%rax)
+0200 [ 7] mov 0xbc8(%r15),%edi
+0207 [ 4] mov (%rsp),%rsi
+020b [ 5] mov $0x8,%edx
+0210 [ 5] call <read@plt>
+0215 [ 4] cmp $0x8,%rax
+0219 [ 2] je <+0x200>
+021b [ 7] mov 0x190(%r15),%rax
+0222 [ 3] inc %r13d
+0225 [ 7] incq 0x440(%rax)
+022c [ 5] jmp <+0xc0>
+0231 [11] data16 cs nopw 0x0(%rax,%rax,1)
+023c [ 4] nopl 0x0(%rax)
+0240 [ 7] mov 0x188(%r15),%rax
+0247 [ 7] movzbl 0x115e0(%rax),%eax
+024e [ 2] cmp $0x3,%al
+0250 [ 6] ja <+0xc0>
+0256 [ 7] mov 0xaa0(%r15),%eax
+025d [ 4] mov %eax,0x10(%rsp)
+0261 [ 2] test %eax,%eax
+0263 [ 6] js <+0xc0>
+0269 [ 8] movl $0x0,0x1c(%rsp)
+0271 [ 4] mov 0x10(%rsp),%edi
+0275 [ 2] xor %edx,%edx
+0277 [ 2] xor %esi,%esi
+0279 [ 5] mov $0x80800,%ecx
+027e [ 5] call <accept4@plt>
+0283 [ 2] test %eax,%eax
+0285 [ 2] jns <+0x2c0>
+0287 [ 5] call <__errno_location@plt>
+028c [ 2] mov (%rax),%eax
+028e [ 3] cmp $0x4,%eax
+0291 [ 2] je <+0x271>
+0293 [ 3] cmp $0xb,%eax
+0296 [ 2] je <+0x2a6>
+0298 [ 7] mov 0x190(%r15),%rax
+029f [ 7] incq 0x458(%rax)
+02a6 [ 4] mov 0x1c(%rsp),%eax
+02aa [ 3] add %eax,%r13d
+02ad [ 5] jmp <+0xc0>
+02b2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+02bd [ 3] nopl (%rax)
+02c0 [ 5] mov $0x1,%edx
+02c5 [ 2] mov %eax,%esi
+02c7 [ 3] mov %r15,%rdi
+02ca [ 4] incl 0x1c(%rsp)
+02ce [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
+02d3 [ 2] jmp <+0x271>
+02d5 [ 5] call <__stack_chk_fail@plt>
